boot/boot_serial/src/boot_serial.c

```diff
@@ -301,7 +301,8 @@ static off_t erase_range(const struct flash_area *fap, off_t start, off_t end)
     }
 
     size = flash_sector_get_off(&sect) + flash_sector_get_size(&sect) - start;
-    BOOT_LOG_INF("Erasing range 0x%x:0x%x", start, start + size - 1);
```

Comment:
Why not to cast this to `ssize` and use `z`?

Reply:
because `ssize_t` is not guaranteed to be the largest integer type supported and/or the same size as `off_t`.

Reply:
> because `ssize_t` is not guaranteed to be the largest integer type supported and/or the same size as `off_t`.

We should probably replace the `off_t`, here,  with the `uint32_t`, including the `erase_range` definition; I do not think that we need offsets and sizes greater than 32-bit here (for example Zephyr flash_map sizes are `uint32_t`). Than just use `llx` and cast expressions to `unsigned long long` - this should cover both 32 and 64 bit systems for us.